modules/oadp-api-details.adoc

```diff
@@ -0,0 +1,240 @@
+// Module included in the following assemblies:
+//
+// * oadp-api.adoc
+
+:_content-type: REFERENCE
+[id="oadp-api-details_{context}"]
+= OADP API
+
+The following tables expose the structure of the OADP API:
+
+.DataProtectionApplicationSpec
+[options="header"]
+|===
+|Property|Type|Description
+
+|`backupLocations`
+|[] link:https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#BackupLocation[`BackupLocation`]
+|Defines the list of configurations to use for `BackupStorageLocations`.
+
+|`snapshotLocations`
+|[] link:https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#SnapshotLocation[`SnapshotLocation`]
+|Defines the list of configurations to use for `VolumeSnapshotLocations`.
+
+|`unsupportedOverrides`
+|map [ link:https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#UnsupportedImageKey[UnsupportedImageKey] ]  link:https://pkg.go.dev/builtin#string[string]
+|Can be used to override the deployed dependent images for development. Options are `veleroImageFqin`, `awsPluginImageFqin`, `openshiftPluginImageFqin`, `azurePluginImageFqin`, `gcpPluginImageFqin`, `csiPluginImageFqin`, `dataMoverImageFqin`, `resticRestoreImageFqin`, `kubevirtPluginImageFqin`, and `operator-type`.
+
+|`podAnnotations`
+|map [ link:https://pkg.go.dev/builtin#string[string] ] link:https://pkg.go.dev/builtin#string[string]
+|Used to add annotations to pods deployed by Operators.
+
+|`podDnsPolicy`
+|link:https://pkg.go.dev/k8s.io/api/core/v1#DNSPolicy[`corev1.DNSPolicy`]
```

Comment:
maybe change the link's text to just `DNSPolicy` instead of corev1.DNSPolicy, as corev1 is not an actual package (it's an import alias). The package is actually v1, but I think it's redundant to mention it.